ts/config.ts

```diff
@@ -73,11 +93,116 @@ export const nody_greeter: nody_config = {
     theme_dir:
       process.env.NODY_GREETER_THEME_DIR || "/usr/share/web-greeter/themes/",
   },
+  theme: {
+    primary_html: "index.html",
+    secondary_html: "",
+  },
 };
 
 const path_to_config =
   process.env.NODY_GREETER_CONFIG || "/etc/lightdm/web-greeter.yml";
 
+let theme_dir: string | undefined;
+
+export function load_theme_dir(): string {
+  const theme = nody_greeter.config.greeter.theme;
+  const dir = nody_greeter.app.theme_dir;
+  const def_theme = "gruvbox";
+  theme_dir = path.join(dir, theme);
+
+  if (theme.startsWith("/")) theme_dir = theme;
+  else if (theme.includes(".") || theme.includes("/"))
+    theme_dir = path.join(process.cwd(), theme);
+
+  if (theme_dir.endsWith(".html")) theme_dir = path.dirname(theme_dir);
+
+  if (!fs.existsSync(theme_dir)) {
+    logger.warn(`"${theme}" theme does not exists. Using "${def_theme}" theme`);
+    theme_dir = path.join(dir, def_theme);
+  }
+
+  return theme_dir;
+}
+
+export function load_primary_theme_path(): string {
+  if (!theme_dir) load_theme_dir();
+  const dir = nody_greeter.app.theme_dir;
+  const def_theme = "gruvbox";
+  const primary = nody_greeter.theme.primary_html;
+  let path_to_theme = path.join(theme_dir, primary);
+
+  if (!path_to_theme.endsWith(".html"))
+    path_to_theme = path.join(path_to_theme, "index.html");
+
+  if (!fs.existsSync(path_to_theme)) {
+    logger.warn(
+      `"${path_to_theme}" theme does not exists. Using "${def_theme}" theme`
+    );
+    path_to_theme = path.join(dir, def_theme, "index.html");
+  }
+
+  nody_greeter.config.greeter.theme = path_to_theme;
+  return path_to_theme;
+}
+export function load_secondary_theme_path(): string {
+  if (!theme_dir) load_theme_dir();
+  const primary = nody_greeter.theme.primary_html;
+  const secondary = nody_greeter.theme.secondary_html;
+  let path_to_theme = path.join(theme_dir, secondary);
+
+  if (!path_to_theme.endsWith(".html"))
+    path_to_theme = path.join(path_to_theme, "index.html");
+
+  if (!fs.existsSync(path_to_theme)) {
+    logger.warn(
+      `"${secondary}" does not exists. Using "${primary}" for secondary monitors`
+    );
+    path_to_theme = load_primary_theme_path();
+  }
+
+  return path_to_theme;
+}
+
+export function load_theme_config(): void {
+  if (!theme_dir) load_theme_dir();
+  const path_to_theme_config = path.join(theme_dir, "index.theme");
+  try {
+    if (!fs.existsSync(path_to_theme_config))
+      throw new Error("Theme config not found");
+
+    const file = fs.readFileSync(path_to_theme_config, "utf-8");
+    const theme_config = yaml.load(file) as theme_config;
+
+    if (!theme_config.primary_html.match(/.*.html/))
+      theme_config.primary_html = "index.html";
+    if (!theme_config.secondary_html.match(/.*.html/))
+      theme_config.secondary_html = "";
```

Comment:
This doesn't accurately check whether the file paths end in `.html`, it will also match things like `foo_html` and `bar.html.something_else`. You'd need to use a regex that includes an end marker and escapes `.` like this: `/^.*\.html$/`. But you could also just use `.endsWith`:

```suggestion
    if (!theme_config.primary_html.endsWith(".html"))
      theme_config.primary_html = "index.html";
    if (!theme_config.secondary_html.endsWith(".html"))
      theme_config.secondary_html = undefined;
```

Reply:
Sure, it wouldn't work xd